src/vmm/src/vstate/vcpu/x86_64.rs

```diff
@@ -291,9 +309,57 @@ impl KvmVcpu {
             vcpu_events,
             xcrs,
             xsave,
+            tsc_khz,
         })
     }
 
+    // Scale the TSC frequency of this vCPU to the one provided as a parameter.
+    pub fn set_tsc_khz(&self, state_tsc: Option<u32>, cpuid: &CpuId) -> Result<()> {
+        // Check if CPU models are the same.
+        if is_same_model(cpuid) {
+            // If they are the same, we don't need to do anything else.
+            return Ok(());
+        }
+
```

Comment:
Should we fail early here if KVM caps for get/set TSC freq are not available ?

Reply:
It is a rather complex state where the `KVM_GET_TSC_KHZ` cap is always needed at this point, but the call is made just below and a failure does not tear the whole restore process down. Furthermore, we will only know if we need the `KVM_SET_TSC_KHZ` cap later, and we check it when we need it. So I don't really see the point of checking the capabilities here.